sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/date/BeamSqlTimestampMinusIntervalExpression.java

```diff
@@ -41,7 +39,7 @@
 
   public BeamSqlTimestampMinusIntervalExpression(
       List<BeamSqlExpression> operands, SqlTypeName outputType) {
-    super(operands, outputType);
+    super(operands, SqlTypeName.TIMESTAMP);
```

Comment:
Seems like outputType will be unused. 

Why do you need to have a fixed parameter `SqlTypeName.TIMESTAMP`? If this is correct then unused function parameter is supposed to be removed.

Reply:
This change is probably just wrong. If this operator was created through `BeamSqlDatetimePlusExpression`, then `outputType` will always be TIMESTAMP. But another code path might create this directly then call `accept` to see if it is valid. This change will make `accept` pass when it should have failed.